### ISSUE TITLE
Auto-refresh build status?

### ISSUE THREAD

**Issue by kengz** [tombell/travis-ci-status#63]
There doesn't seem to be an option to manually fetch the build status?
I pushed my git and rebuilt travis to passing, and after 1 minute the clock-icon in my Atom is still red. Had to restart Atom for it to turn green.

How often is it refreshed?

**Comment by tombell** [comment]
As far as I remember, it doesn't refresh. Would accept a pull request to add an Atom command to refresh it.

**Comment by CodingAnarchy** [comment]
It looks like [this code here](https://github.com/tombell/travis-ci-status/blob/master/lib/build-status-view.coffee) has an attempt to get it to update whenever the GitHub repo changes, but it doesn't match the Atom API from what I can tell.

You have:

```coffeescript

    repos = atom.project.getRepositories()
    console.log "DEBUG:", repos
    name = atom.config.get('travis-ci-status.travisCiRemoteName')
    repo = repos.filter((r) -> /(.)*github\.com/i.test(r.getConfigValue("remote.#{name}.url")))
    @repo = repo[0]

    $(@repo).on 'status-changed', (path, status) =>
      @update() if path is @getActiveItemPath()

    $(@repo).on 'statuses-changed', @update
    @update()
```

However, [the API refers to a function](https://atom.io/docs/api/v1.0.0/GitRepository#instance-onDidChangeStatuses): `::onDidChangeStatuses(callback)` which operates this in the background: `@emitter.on 'did-change-statuses', callback`.

There appear to be some updates to the manner in which Atom prefers you get the repositories (atom.project.getRepositories() is deprecated for removal in 2.0, with the replacement of `Promise.all(atom.project.getDirectories().map(
    atom.project.repositoryForDirectory.bind(atom.project)))` being desired).

In short, this should be fixable with some updates for Atom v1.0.0.  I'd do it myself, but I'm not really an expert with CoffeeScript.

**Comment by Sgoettschkes** [comment]
I'm :+1: on this. It's not really useful if it does not auto-refresh or at least offer a way to manually refresh. I'd give it a go but have never worked on atom packages before and currently do not have the time.

**Comment by CodingAnarchy** [comment]
I have a preliminary version that I think will modify the determination of the repository for the current version of Atom and automatically update the Travis CI status when the files in the git project have changed.

CodingAnarchy/travis-ci-status@00eb63b

If anyone wants to go through and test this additionally, I'd appreciate it.  It passes the tests that are in the repository, but I'm less confident in my testing that validates the refreshing is occurring as expected.

**Comment by tombell** [comment]
@CodingAnarchy when feel feel it's ready, open and PR and I'll merge and release.

**Comment by CodingAnarchy** [comment]
It looks like I need to do some further investigation.  The promise appears to work for fetching the repository info, but I'm still not getting an update when the files change on disk - or any other time, for that matter.  The current commit replicates the existing behavior, though so it might be useful as an update just to ensure the old method isn't dropped due to a new version release.  I'll make a PR for that.

**Comment by CodingAnarchy** [comment]
This should be somewhat sorted out by #74, as that will result in a refresh of the status whenever the git repository on the disk has a status change.  Once that is released, this should be able to be closed unless there is some more desirable behavior that is really wanted.
